code/game/objects/structures/crates_lockers/closets/bsdroppod.dm

```diff
@@ -0,0 +1,89 @@
+//The "BDPtarget" temp visual is created by the expressconsole, which in turn makes two things: a falling droppod animation, and the droppod itself.
+
+
+//------------------------------------BLUESPACE DROP POD-------------------------------------//
+/obj/structure/closet/bsdroppod
+	name = "Bluespace Drop Pod"
+	desc = "A Nanotrasen supply drop pod."
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP"
+	pixel_x = -16//2x2 sprite
+	pixel_y = -5
+	pixel_z = 0
+	layer = TABLE_LAYER//so that the crate inside doesn't appear underneath
+	allow_objects = TRUE
+	allow_dense = TRUE
+	delivery_icon = null
+	can_weld_shut = FALSE
+	armor = list(melee = 30, bullet = 50, laser = 50, energy = 100, bomb = 90, bio = 0, rad = 0, fire = 80, acid = 80)
+	anchored = TRUE
+	anchorable = FALSE
+	var/datum/supply_order/SupplyOrder
+
+/obj/structure/closet/bsdroppod/Initialize(mapload, datum/supply_order/so)
+	. = ..()
+	src.SupplyOrder = so//uses Supply Order passed from expressconsole into BDPtarget
+	addtimer(CALLBACK(src, .proc/open), 30)//open 3seconds after appearing
+
+/obj/structure/closet/bsdroppod/update_icon()//called in initialize, all we have to do is add a door icon
+	cut_overlays()//just incase update_icon somehow gets called multiple times, prevents overlay spam
+	add_overlay("BDP_door")
+
+/obj/structure/closet/bsdroppod/proc/sparks()
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, get_turf(src))
+	s.start()
+	qdel(src)
+
+/obj/structure/closet/bsdroppod/attackby(obj/item/W, mob/user, params)//i no need to be welded or anchored or whatever
+	return
+
+/obj/structure/closet/bsdroppod/toggle(mob/living/user)//ditto
+	return
+
+/obj/structure/closet/bsdroppod/open()
+	var/turf/T = get_turf(src)
+	SupplyOrder.generate(T)//not called during populateContents as supplyorder generation requires a turf
+	cut_overlays()
+	add_overlay("BDP_open")
+	playsound(loc, open_sound, 15, 1, -3)
+	addtimer(CALLBACK(src, .proc/sparks), 30)
+
+//------------------------------------FALLING BLUESPACE DROP POD-------------------------------------//
+/obj/effect/temp_visual/BDPfall
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP_falling"
+	pixel_x = -16
+	pixel_y = -5
+	pixel_z = 200
+	name = "Bluespace Drop Pod"
+	desc = "Get out of the way!"
+	layer = FLY_LAYER//that wasnt flying, that was falling with style!
+	randomdir = FALSE
+
+/obj/effect/temp_visual/BDPfall/Initialize()//prevents QDEL_IN timer creation
+	return
```

Comment:
Nononono, if you want to prevent that timer from being created, refactor it out into a seperate proc and override it. Initialize always has to go all the way

Reply:
refactor temp_visual?

Reply:
yes, alternatively you could use just /obj/effect but I'll leave that up to you

Reply:
done